File: GuideObjC/GuideObjC/ViewControllers/InitController/InitController.m
//
//  InitController.m
//  GuideObjC
//
//  Created by LuzanovRoman on 09.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

#import "InitController.h"
#import "OffensivePosts.h"
#import "Follows.h"

@interface InitController ()

@end

@implementation InitController

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    if (self.graph.isAuthorized) {
        [[Follows shared] startObserving];
        [[OffensivePosts shared] startSession];
        [self performSegueWithIdentifier:@"ShowTabBar" sender:nil];
    }
    else {
        [self performSegueWithIdentifier:@"ShowLoginScreen" sender:nil];
    }
}
@end

File: GuideSwift/GuideSwift/ViewControllers/InitController/InitController.swift
//
//  InitController.swift
//  GuideSwift
//
//  Created by LuzanovRoman on 09.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

import UIKit

class InitController: UIViewController {

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        if Graph.isAuthorized {
            Follows.shared.startObserving()
            OffensivePosts.shared.startSession()
            performSegue(withIdentifier: "ShowTabBar", sender: nil)
        } else {
            performSegue(withIdentifier: "ShowLoginScreen", sender: nil)
        }
    }
}
